examples/metrics-advanced/src/main.rs

```diff
@@ -49,13 +49,14 @@ fn init_meter_provider() -> opentelemetry_sdk::metrics::SdkMeterProvider {
         .with_temporality(Temporality::Delta)
         .build();
 
+    let resource = Resource::builder()
+        .with_key_value(KeyValue::new("service.name", "metrics-advanced-example"))
+        .build();
+
     let reader = PeriodicReader::builder(exporter, runtime::Tokio).build();
     let provider = SdkMeterProvider::builder()
         .with_reader(reader)
-        .with_resource(Resource::new([KeyValue::new(
-            "service.name",
-            "metrics-advanced-example",
-        )]))
+        .with_resource(resource)
```

Comment:
wondering if we can do `with_servicename_resource("metrics-advanced-example")` ? This will be the most common use case.
Or just `Resource::builder().with_servicename("metrics-advanced-example")......build()` ?

Reply:
I like that, I was using the go implementation for Resource builder this weekend and they had something similar to this and it felt pretty nice
```go
sResource, err := resource.New(
  ctx,
  resource.WithTelemetrySDK(),
  resource.WithAttributes(semconv.ServiceName("trace-export-service")),
)
```

Reply:
added the option of `.with_service_name("name")`

Reply:
Hmm should we give special treatment to service name? I feel like it will open a can of worms that which attributes should get dedicate builder method. I think we should keep the API lean for now. 

On the long term we can use dedicate trait to enrich the builder. Image something like 
```rust
pub trait SemCov {
  fn with_service_name(&str);
  fn with_service_version(&str);
  ...
  // those method can probably generated in sem cov crate
}

impl SemCov for ResourceBuilder {
  fn with_service_name(&str) {
    ...
   // again we can generated those code in sem cov crate
  }
}

// main.rs
import {ResourceBuilder, SemCov}

let builder = ResourceBuilder::new().with_service_name("test")
```

Reply:
Sounds reasonable to me, I was kind of wondering the same thing where we would want to use the key from sem conv instead of the hardcoded one in the sdk. I will remove this, and it can become another discussion

Reply:
so do we want to add the option for service.name?

Reply:
Yes let's do it. Its okay to do it as a separate follow up PR so this PR is not blocked.

Reply:
added it back because its small